File: tutorials/code/simp/python/eval-pregenerated/eval-pregenerated-print.cpp
#include <linkpred.hpp>
#include <iostream>
using namespace LinkPred::Simp;
int main() {
	// Create an evaluator object
	Evaluator eval;
	// Add predictors to be evaluated
	eval.addADA();
	eval.addRAL();
	// Add performance measures
	eval.addPR();
	eval.addTPR();
	// Run experiment on the specified network
	eval.run("Zakarays_Karate_Club_Train.edges", "Zakarays_Karate_Club_Test.edges");
	auto res = eval.getPerfRes(0);
	for (auto it =  res.begin(); it != res.end(); ++it) {
		std::cout << it->name << "\t" ;
	}
	std::cout << "\n";
	for (auto it =  res.begin(); it != res.end(); ++it) {
		std::cout << it->res << "\t" ;
	}
	std::cout << "\n";
	return 0;
}

File: tutorials/code/simp/python/eval-pregenerated/eval-pregenerated-print.py
# Import the module
import LinkPredPython as lpp
# Create an evuator object
ev = lpp.Evaluator();
# Add predictors to be evuated
ev.addADA();
ev.addRAL();
# Add performance measures
ev.addPR();
ev.addTPR();
# Run experiment on the specified network
ev.run("Zakarays_Karate_Club_Train.edges", "Zakarays_Karate_Club_Test.edges");
import sys # For printing
res = ev.getPerfRes(0);
for r in res:
	sys.stdout.write(r.name + "\t");
sys.stdout.write("\n");
for r in res:
	sys.stdout.write("{:.4f}".format(r.res) + "\t");
sys.stdout.write("\n");
